### PR TITLE
Introduction to koa.js

### PR DESCRIPTION
**Pull request by kayere**
Added Introduction to Koa.js Article.

### CODE REVIEW

File: articles/introduction-to-koa.js/index.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/introduction-to-koajs/hero.jpg
+    alt: koa koajs javascript JavaScript node nodejs node.js framework
+---
+In my previous article, an introduction to the [hapi.js framework](https://www.section.io/engineering-education/introduction-to-hapi/), I mentioned [Koa.js](https://www.koajs.com), another major node.js framework. This article will give an introduction to the framework, its features and a tutorial of how to create a simple server using this framework.
+<!--more-->
+### What is Koa.js?
+Koa.js is an open source node.js web framework developed and maintained by the team behind express. This framework was build with the aim of making a smaller, more expressive and more robust foundations for web applications and Apis. Koa uses asycn functions that help eliminate callbacks and greatly increase error handling. Koa does not bundle midlewares within its core, and provides a refined suite of methods that make writing servers fast and exciting.
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:20]: of making a smaller, more  -> of making smaller, more

Comment (nadivgold) [reply articles/introduction-to-koa.js/index.md:20]: applications and Apis. Koa ->  applications and APIs. Koa

Comment (nadivgold) [reply articles/introduction-to-koa.js/index.md:20]: Koa uses asycn functions ->  Koa uses asynchronous functions

Comment (nadivgold) [reply articles/introduction-to-koa.js/index.md:20]: functions that help eliminate callbacks -> functions to help eliminate the need for callbacks

File: articles/introduction-to-koa.js/index.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+### Koa.js features
+As it is with all frameworks available, Koa.js also has its own unique features that make it work as intended by the developers.
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:23]: it is with all frameworks available, Koa.js also ->  it is with all available frameworks, Koa.js also

Comment (nadivgold) [reply articles/introduction-to-koa.js/index.md:23]: delete: that make it work as intended by the developers.  

replace with the reason why these unique features are interesting/more productive/better practice/etc.

File: articles/introduction-to-koa.js/index.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    alt: koa koajs javascript JavaScript node nodejs node.js framework
+---
+In my previous article, an introduction to the [hapi.js framework](https://www.section.io/engineering-education/introduction-to-hapi/), I mentioned [Koa.js](https://www.koajs.com), another major node.js framework. This article will give an introduction to the framework, its features and a tutorial of how to create a simple server using this framework.
+<!--more-->
+### What is Koa.js?
+Koa.js is an open source node.js web framework developed and maintained by the team behind express. This framework was build with the aim of making a smaller, more expressive and more robust foundations for web applications and Apis. Koa uses asycn functions that help eliminate callbacks and greatly increase error handling. Koa does not bundle midlewares within its core, and provides a refined suite of methods that make writing servers fast and exciting.
+
+### Koa.js features
+As it is with all frameworks available, Koa.js also has its own unique features that make it work as intended by the developers. 
+Below is a highlight of some of the features.
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:24]: Below is a highlight of some of the features. -> Here are a highlight of some of the features in Koa:

File: articles/introduction-to-koa.js/index.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+1. Koa.js is modern and futuristic - Unlike other Node.js frameworks, Koa.js is built based on ECMAScript 2015 specification which simplifies development of complex applications by providing a bunch of new classes and modules. This helps developers to create maintainable applications.
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:25]: modern and futuristic - Unlike ->  modern and future-proof - Unlike

*is this what you meant?

**Comment by hectorkambow** [comment]
@kayere - after running your Koa.js article through our 3rd party plagiarism checker - it raised a few red flags. 
Some of which were not valid - but a few that did seem a bit too close - to not include a citation if you are paraphrasing. 
Always best to be safe than sorry - 
See attached - 
Thank you 
[2120420-2d4acd820416a38f3101-koa_intro.pdf](https://github.com/section-io/engineering-education/files/5153782/2120420-2d4acd820416a38f3101-koa_intro.pdf)

**Comment by kayere** [comment]
Okay @hectorkambow and @nadivgold. Let me make the necessary changes.

**Comment by kayere** [comment]
Hello @hectorkambow I have made the changes. Please review. Thanks

File: articles/introduction-to-koa.js/index.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    alt: koa koajs javascript JavaScript node nodejs node.js framework
+---
+In my previous article, about an introduction to [hapi.js framework](https://www.section.io/engineering-education/introduction-to-hapi/), I mentioned [Koa.js](https://www.koajs.com), another major node.js framework. This article will give an introduction to the framework, its features and a tutorial of how to create a simple server using this framework.
+<!--more-->
+### What is Koa.js?
+Koa.js is an open source node.js web framework that was designed by the team behind express. As their [official website](https://koajs.com) says, the framework aims to be a smaller, more expressive, and more robust foundation for web applications and APIs. Koa uses asynchronous functions to help eliminate the need for callbacks and significantly improve error handling. Koa does not bundle middlewares within its core, and provides a refined suite of methods that fasten the process of making servers and makes it exciting.
+
+### Koa.js features
+As it is with all available frameworks, Koa.js also has its own unique features that make it more expressive and developer-friendly.
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:23]: As it is with all -> As with all

File: articles/introduction-to-koa.js/index.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+Here are a highlight of some of the features in Koa.
+1. Koa.js is modern and future-proof - Unlike other Node.js frameworks, Koa.js build is based on ES6 which makes development of complex applications simpler by providing a bunch of new classes and modules. This helps developers to create maintainable applications.
+2. Koa.js uses ES6 generators to simplify synchronous programming and facilitate flow of controls. This generators can also be used as functions to control code execution on the same stack
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:26]: controls. This generators  -> controls. These generators

File: articles/introduction-to-koa.js/index.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+```js
+const koa = require('koa')
+const app = new koa()
+
+app.listen(2400, () => {console.log('Server running at PORT 2400')})
+```
+The code used to create the server is pretty straight forward, just import the koa module and use its listen method. To start the server, run node 'name of your index file' on the terminal.
+
+### creating routes using koa.js
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:67]: Capitalize subheadings

File: articles/introduction-to-koa.js/index.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+})
+
+app.use(router.routes())
+  .use(router.allowedMethods())// registering routes to the application
+
+app.listen(2400, () => console.log('Server running at PORT 2400'))
+```
+Start the server again and test the route by sending request from the browser.
+
+### Handling responses in Koa.js
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:96]: Capitalize responses

File: articles/introduction-to-koa.js/index.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+As we mentioned earlier, Koa response objects are embedded in its context object. This means that we access the response object from the context object.
+Lets use a route definition like the one above to demonstrate handling responses.
+
+```js
+router.get('home', '/', (context) => {
+  context.status = 200 //This is the response status
+  coontext.body   = "Welcome to my Koa.js Server" // This is the response body
```
Comment (nadivgold) [inline articles/introduction-to-koa.js/index.md:103]: coontext.body ->  context.body

**Comment by kayere** [comment]
@nadivgold - I have made the changes

**Pull request merged**